src/impl/android/external_commit_helper.cpp

```diff
@@ -106,7 +106,18 @@ ExternalCommitHelper::ExternalCommitHelper(RealmCoordinator& parent)
         }
         // the fifo already existing isn't an error
         if (ret == -1 && err != EEXIST) {
-            throw std::system_error(err, std::system_category());
+            // Workaround for a mkfifo bug on Blackberry devices:
+            // When the fifo already exists, mkfifo fails with error ENOSYS which is not correct.
```

Comment:
Right, I'll fix that.

One concern is that the code is from https://github.com/realm/realm-core/blob/v2.1.0/src/realm/util/interprocess_condvar.cpp#L100 and I can't test that.